src/main/java/com/rackspace/ceres/app/services/MetadataService.java

```diff
@@ -213,7 +223,36 @@ public MetadataService(ReactiveCqlTemplate cqlTemplate,
         )
         .flatMapMany(Flux::fromIterable);
   }
+  
+  public Flux<TsdbQuery> getMetricsAndTagsAndMetadata(List<TsdbQueryRequest> queries, List<Granularity> granularities) {
+    List<TsdbQuery> result = new ArrayList<TsdbQuery>();
+    
+    for (TsdbQueryRequest query : queries) {
+      List<Map<String, String>> filters = query.getFilters();
+      
+      for (Map<String, String> filter : filters) {
+        String[] splitValues = filter.get("filter").split("\\|");
+        for (int i = 0; i < splitValues.length; i++) {
 
+          Map<String, String> tags = new HashMap<String, String>();
+          tags.put(filter.get("tagk"), splitValues[i]);
+                    
+          String downsample = query.getDownsample();
+          String[] values = downsample.split("-");
```

Comment:
Same here with '-'?

Reply:
Need to parse out the granularity width and aggregator